content/en/blog/_posts/2023-09-13-userns-stateful-pods/index.md

```diff
@@ -0,0 +1,148 @@
+`UserNamespacesSupport`. There were many changes done and the requirements on
+the node hosts changed. So with Kubernetes 1.28 the feature flag was renamed to
+reflect this.
+
+## Demo
+
+Rodrigo created a demo which exploits [CVE 2022-0492][cve-link] and shows how
+the exploit can occur without user namespaces. He also shows how it is not
+possible to use this exploit from a Pod where the containers are using this
+feature.
+
+This vulnerability is rated **HIGH** and allows **a container with no special
+privileges to read/write to any path on the host** and launch processes as root
+on the host too.
+
+{{< youtube id="M4a2b4KkXN8" title="Mitigation of CVE-2022-0492 on Kubernetes by enabling User Namespace support">}}
+
+Most applications in containers run as root today, or as a semi-predictable
+non-root user (user ID 65534 is a somewhat popular choice). When you run a Pod
+with containers using a userns, Kubernetes runs those containers as unprivileged
+users, with no changes needed in your app.
+
+This means two containers running as user 65534 will effectively be mapped to
+different users on the host, limiting what they can do to each other in case of
+an escape, and if they are running as root, the privileges on the host are
+reduced to the one of an unprivileged user.
+
+[cve-link]: https://unit42.paloaltonetworks.com/cve-2022-0492-cgroups/
+
+## Node system requirements
+
+There are requirements on the Linux kernel version as well as the container
+runtime to use this feature.
+
+On Linux you need Linux 6.3 or greater. This is because the feature relies on a
+kernel feature named idmap mounts, and support to use idmap mounts with tmpfs
+was merged in Linux 6.3.
+
+If you are using CRI-O with crun, this is [supported in CRI-O
+1.28.1][CRIO-release] and crun 1.8.8 or greater. If you are using CRI-O with runc,
+this is still not supported.
+
+containerd support is currently targeted for containerd 2.0; it is likely that
+it won't matter if you use it with crun or runc.
+
+Please note that containerd 1.7 added _experimental_ support for user
+namespaces as implemented in Kubernetes 1.25 and 1.26. The redesign done in 1.27
+is not supported by containerd 1.7, therefore it only works, in terms of user
+namespaces support, with Kubernetes 1.25 and 1.26.
+
+One limitation present in containerd 1.7 is that it needs to change the
+ownership of every file and directory inside the container image, during Pod
+startup. This means it has a storage overhead and can significantly impact the
+container startup latency. Containerd 2.0 will probably include a implementation
+that will eliminate the startup latency added and the storage overhead. Take
+this into account if you plan to use containerd 1.7 with user namespaces in
+production.
+
+None of these containerd limitations apply to [CRI-O 1.28][CRIO-release].
+
+[CRIO-release]: https://github.com/cri-o/cri-o/releases/tag/v1.28.1
```

Comment:
Talking with @haircommander we will make this release before this blog is published, and we prefer to reference this release.

Reply:
The link already works! :)